nixos/modules/services/hardware/undervolt.nix

```diff
@@ -165,7 +177,7 @@ in
 
     environment.systemPackages = [ cfg.package ];
 
-    systemd.services.undervolt = {
+    systemd.services.undervolt = mkIf hasCliArgs {
```

Comment:
```suggestion
    systemd.services.undervolt = mkIf (cliArgs != [ ]) {
```

Reply:
I am afraid this will not work.  If users only set `services.undervolt.verbose = true` but no other settings, then `cliArgs` will be `[ "--verbose" ]`

Reply:
eh fair enough, I still feel it's redundant to list out every single flag though. Maybe exclude the flags that don't allow the service to start?

Reply:
It's redundant, but this is one of the times where a tradeoff has to be made between correct functionality and pretty code. This code, while not pretty, is self-evident. Having a list of flags be *excluded* will result in harder-to-read code that looks marginally prettier.

Reply:
It's not about "pretty" but "error-prone".

Reply:
How is what I have error-prone?

Reply:
You don't think maintaining a list of every single flag by hand is error prone?

Reply:
> Maintaining a list of every single flag by hand is error prone

This is true.  However, I guess we have to do this for undervolt.  Both of your proposals are essentially doing so.  The difference is maintaining every "important" flags or maintaining every "unimportant" flags like `verbose`.  I do not have a strong opinion on this.

Reply:
> Maintaining a list of every single flag by hand is error prone.

No it's not?

Reply:
> However, I guess we have to do this for undervolt. Both of your proposals are essentially doing so. The difference is maintaining all "important" flags or maintaining all "unimportant" flags like `verbose`.

You're correct. Here's my reasoning - if we maintain an inclusion list manually, and we miss adding a flag, then the result is that the module is unusable (i.e. this blocks rebuilds) unless users are copypasting and replacing the module entirely.

If we maintain an _exclusion_ list, and the exclusion list is missing an entry, the failure mode is that the service doesn't start - which can be fixed immediately by reading upstream docs and creating a GH issue/PR to add the exclusion entry.

NB I am making the assumption that it's more likely to forget to add a new flag to either type of list, than to incorrectly add it.

Reply:
> The service will fail to start if no undervolt settings are applied. This disables the service to not pollute the journal with failures.

This is the original version of PR description and probably is the author's motivation.

I am a bit confused about how this service pollutes the journal with failures.  There is no restart logic defined for this service so it only produces **one** failure log after `nixos-rebuild test (or switch)`.

I am not a fan of maintaining a list in assertion.  IMHO, in this case, a runtime failure signal from `nixos-rebuild` is better than an eval time error (assertion).

Reply:
> I am a bit confused about how this service pollutes the journal with failures. There is no restart logic defined for this service so it only produces one failure log after nixos-rebuild test (or switch).

I reboot frequently.

Reply:
> > I am a bit confused about how this service pollutes the journal with failures. There is no restart logic defined for this service so it only produces one failure log after nixos-rebuild test (or switch).
> 
> I reboot frequently.

I do not follow.  Why don't you fix your config for `services.undervolt` when `nixos-rebuild` gives you an error?

Reply:
`nixos-rebuild boot` doesn't give an error though? It evals and builds the system just fine.